lib/hdt.js

```diff
@@ -5,53 +5,78 @@ var hdtNative = require('../build/Release/hdt');
 var HdtDocumentPrototype = hdtNative.HdtDocument.prototype;
 
 // Searches the document for triples with the given subject, predicate, and object.
-HdtDocumentPrototype.searchTriples = function (subject, predicate, object, options, callback, self) {
-  if (typeof  callback !== 'function') self = callback, callback = options, options = {};
-  if (typeof  callback !== 'function') return;
-  if (this.closed) return callback.call(self || this, new Error('The HDT document cannot be read because it is closed'));
+HdtDocumentPrototype.searchTriples = function (subject, predicate, object, options) {
+  if (this.closed) return Promise.reject(new Error('The HDT document cannot be read because it is closed'));
   if (typeof   subject !== 'string' ||   subject[0] === '?') subject   = '';
   if (typeof predicate !== 'string' || predicate[0] === '?') predicate = '';
   if (typeof    object !== 'string' ||    object[0] === '?') object    = '';
-  var offset = options && options.offset ? Math.max(0, parseInt(options.offset, 10)) : 0,
+  const offset = options && options.offset ? Math.max(0, parseInt(options.offset, 10)) : 0,
       limit  = options && options.limit  ? Math.max(0, parseInt(options.limit,  10)) : 0;
 
-  this._searchTriples(subject, predicate, object, offset, limit, callback, self);
+  return new Promise((resolve, reject) => {
+    this._searchTriples(subject, predicate, object, offset, limit, function (err, triples, totalCount, hasExactCount) {
+      if (err) return reject(err);
+
+      return resolve({
+        triples,
+        totalCount,
+        hasExactCount,
+      });
+    }, this);
+  });
 };
 
 // Gives an approximate number of matches of triples with the given subject, predicate, and object.
-HdtDocumentPrototype.countTriples = function (subject, predicate, object, callback, self) {
-  this.search(subject, predicate, object, { offset: 0, limit: 0 },
-              function (error, triples, totalCount, hasExactCount) {
-                callback.call(this, error, totalCount, hasExactCount);
-              }, self);
+HdtDocumentPrototype.countTriples = function (subject, predicate, object) {
+  return this.search(subject, predicate, object, { offset: 0, limit: 0 })
+    .then(results => ({ totalCount: results.totalCount, hasExactCount: results.hasExactCount }));
```

Comment:
Maybe remove the `then` part? Doesn't matter that there's an extra `triples` field (plus, fewer object shapes for V8).